core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorTableLayout.java

```diff
@@ -35,8 +35,14 @@ public ConnectorTableLayout(ConnectorPartitioningHandle partitioning, List<Strin
      */
     public ConnectorTableLayout(List<String> partitionColumns)
     {
-        this.partitioning = Optional.empty();
+        this(Optional.empty(), partitionColumns, true);
+    }
+
+    public ConnectorTableLayout(Optional<ConnectorPartitioningHandle> partitioning, List<String> partitionColumns, boolean multipleWritersPerPartitionSupported)
```

Comment:
This constructor is too confusing. Let's just keep:
```
ConnectorTableLayout(ConnectorPartitioningHandle partitioning, List<String> partitionColumns, boolean multipleWritersPerPartitionSupported)
ConnectorTableLayout(ConnectorPartitioningHandle partitioning, List<String> partitionColumns)
ConnectorTableLayout(List<String> partitionColumns)
```

In fact, we could consider introducing static factory methods (in a backward compatible way)